### PR TITLE
WT-7802 Remove data store same transaction update squash logic

### PR DESCRIPTION
**Pull request by kommiharibabu**
The logic of stable update in RTS and reconciliation select is
diverged due to the data store squash logic. The history store
properly handles the squash logic, there is no point in having
it multiple places. Removing it from the reconciliation select makes
the code in sync with RTS.

### CODE REVIEW

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -488,48 +464,14 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
 
             /* Find the update this tombstone applies to. */
             if (!__wt_txn_upd_visible_all(session, upd)) {
-                upd = __get_valid_upd(upd, tombstone, &same_txn_valid_upd);
+                while (upd->next != NULL && upd->next->txnid == WT_TXN_ABORTED)
```
Comment (hbokhari) [inline src/reconcile/rec_visibility.c:467]: Tip- maybe write this as:
`for (; upd->next != NULL && upd->next->txnid == WT_TXN_ABORTED; upd = upd->next;)`

Ignore this if you like it the way it's written.

File: src/reconcile/rec_visibility.c [hunk 2]
```diff
@@ -488,48 +464,14 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
...
-                     */
-                    WT_ASSERT(session, same_txn_valid_upd->type != WT_UPDATE_TOMBSTONE);
-                    upd_select->upd = upd = same_txn_valid_upd;
-                }
             }
         }
+
         if (upd != NULL)
             /* The beginning of the validity window is the selected update's time point. */
             WT_TIME_WINDOW_SET_START(select_tw, upd);
```
Comment (hbokhari) [inline src/reconcile/rec_visibility.c:477]: Update the comment below (sorry but GitHub won't let me add a comment on the exact line. ):            
           /*
             * We only have a tombstone on the update list or all the updates are from the same
             * transaction.
             */

File: test/suite/test_rollback_to_stable13.py [hunk 3]
```diff
@@ -183,3 +183,73 @@ def test_rollback_to_stable_with_aborted_updates(self):
         stat_cursor = self.session.open_cursor('statistics:', None, None)
         restored_tombstones = stat_cursor[stat.conn.txn_rts_hs_restore_tombstones][2]
         self.assertEqual(restored_tombstones, nrows)
+
+    def test_rollback_to_stable_with_history_tombstone(self):
+        nrows = 1000
+
+        # Prepare transactions for column store table is not yet supported.
+        if self.prepare and self.key_format == 'r':
```
Comment (hbokhari) [inline test/suite/test_rollback_to_stable13.py:191]: Is it possible to add prepared transactions to this test?

Comment (kommiharibabu) [reply test/suite/test_rollback_to_stable13.py:191]: There is already some test to describe the same transaction prepared updates in test_prepare08.py cover the scenario.

File: src/reconcile/rec_visibility.c (hunk 1 above)
Comment (kommiharibabu) [reply src/reconcile/rec_visibility.c:467]: As there is no performance optimization with the suggested change, I feel the existing change may be more readable. I fine to change if others also feel the same.

File: src/reconcile/rec_visibility.c (hunk 2 above)
Comment (kommiharibabu) [reply src/reconcile/rec_visibility.c:477]: Fixed.

File: src/reconcile/rec_visibility.c [hunk 4]
```diff
@@ -488,56 +464,19 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
...
         else if (select_tw->stop_ts != WT_TS_NONE || select_tw->stop_txn != WT_TXN_NONE) {
-            /*
-             * We only have a tombstone on the update list or all the updates are from the same
-             * transaction.
-             */
+            /* We only have a tombstone on the update list. */
             WT_ASSERT(session, tombstone != NULL);
 
             /* We must have an ondisk value and it can't be a prepared update. */
```
Comment (hbokhari) [inline src/reconcile/rec_visibility.c:482]: I believe there is another similar looking comment further down. Can you fix it as well?

                /*
                 * Tombstone is the only non-aborted update on the update chain or all the updates
                 * are from the same transaction.
                 */

Comment (kommiharibabu) [reply src/reconcile/rec_visibility.c:482]: Thanks for spotting it. Fixed and also updated the test_prepare08 to write prepared updates to disk with cursor reset.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by hbokhari — APPROVED** [review]
LGTM.

**Pull request merged**